File: backend/models/user.model.js

import mongoose from "mongoose";

const user = mongoose.Schema(
    {
        userName: {
            type: String,
            required: [true, "Username is required."],
        },
        password: {
            type: String,
            required: [true, "Password is required."]
        },
        email: {
            type: String,
            required: [true, "Email is required."],
        },
    }
);


export default mongoose.model("User", user);

File: mobile/lib/features/auth/data/models/user_model.dart
import 'package:mobile/features/auth/domain/entities/user.dart';

class UserModel extends UserEntity {
  const UserModel({
    required super.userName,
    required super.email,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(userName: json['userName'], email: json['email']);
  }

  Map<String, dynamic> toJson() {
    return {
      'userName': userName,
      'email': email,
    };
  }
}
